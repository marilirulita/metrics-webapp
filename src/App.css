body {
  color: blue;
}

h2 {
  margin: 20px;
}

h4, 
h3 {
  margin: 0;
}

.App {
  display: flex;
  flex-direction: column;
  width: min-content;
  max-width: 500px;
  margin: 0 auto;
}

nav {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
  padding: 5px;
  color: blue;
}

nav * {
  display: flex;
}

nav span {
  padding: 5px;
}

nav svg {
  color: blue;
  padding: 0;
  height: 100%;
}

.home-container {
  display: flex;
  flex-direction: column;
}

.header-style {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid gray;
  padding: 10px;
};

.header-style-chil {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid gray;
  padding: 10px;
};

.element-text-style {
  display: flex;
  flex-direction: column;
  text-align: right;
  margin: 10px 0;
};