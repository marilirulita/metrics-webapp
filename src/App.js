import {
  Routes, Route, Link, useLocation,
} from 'react-router-dom';
import { RiArrowLeftSLine } from 'react-icons/ri';
import { BiMicrophone } from 'react-icons/bi';
import { FiSettings } from 'react-icons/fi';
import { useEffect } from 'react';
import Home from './components/Home';
import Details from './components/Details';
import './App.css';

function App() {
  const location = useLocation();
  useEffect(() => {
  }, [location]);

  return (
    <div className="App">
      <nav>
        <Link to="/"><RiArrowLeftSLine size="35px" color="white" /></Link>
        <h1>{location.pathname === '/details' ? 'Company Stock' : 'Higher Market Cap'}</h1>
        <div>
          <span><BiMicrophone size="25px" color="white" /></span>
          <span><FiSettings size="25px" color="white" /></span>
        </div>
      </nav>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="details" element={<Details />} />
      </Routes>
    </div>
  );
}

export default App;
