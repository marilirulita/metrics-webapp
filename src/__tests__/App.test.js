import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <App />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
