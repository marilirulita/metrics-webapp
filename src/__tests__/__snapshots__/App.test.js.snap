// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        color="white"
        fill="currentColor"
        height="35px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 24 24"
        width="35px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.828 12l4.95 4.95-1.414 1.414L8 12l6.364-6.364 1.414 1.414z"
          />
        </g>
      </svg>
    </a>
    <h1>
      Higher Market Cap
    </h1>
    <div>
      <span>
        <svg
          color="white"
          fill="currentColor"
          height="25px"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width="25px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 12V6c0-2.217-1.785-4.021-3.979-4.021a.933.933 0 0 0-.209.025A4.006 4.006 0 0 0 8 6v6c0 2.206 1.794 4 4 4s4-1.794 4-4zm-6 0V6c0-1.103.897-2 2-2a.89.89 0 0 0 .163-.015C13.188 4.06 14 4.935 14 6v6c0 1.103-.897 2-2 2s-2-.897-2-2z"
          />
          <path
            d="M6 12H4c0 4.072 3.061 7.436 7 7.931V22h2v-2.069c3.939-.495 7-3.858 7-7.931h-2c0 3.309-2.691 6-6 6s-6-2.691-6-6z"
          />
        </svg>
      </span>
      <span>
        <svg
          color="white"
          fill="none"
          height="25px"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width="25px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </span>
    </div>
  </nav>
  <div
    className="home-container"
  >
    <div
      className="header-style"
    >
      <span>
        <svg
          color="#b13967"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#b13967",
            }
          }
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M621.16 54.46C582.37 38.19 543.55 32 504.75 32c-123.17-.01-246.33 62.34-369.5 62.34-30.89 0-61.76-3.92-92.65-13.72-3.47-1.1-6.95-1.62-10.35-1.62C15.04 79 0 92.32 0 110.81v317.26c0 12.63 7.23 24.6 18.84 29.46C57.63 473.81 96.45 480 135.25 480c123.17 0 246.34-62.35 369.51-62.35 30.89 0 61.76 3.92 92.65 13.72 3.47 1.1 6.95 1.62 10.35 1.62 17.21 0 32.25-13.32 32.25-31.81V83.93c-.01-12.64-7.24-24.6-18.85-29.47zM48 132.22c20.12 5.04 41.12 7.57 62.72 8.93C104.84 170.54 79 192.69 48 192.69v-60.47zm0 285v-47.78c34.37 0 62.18 27.27 63.71 61.4-22.53-1.81-43.59-6.31-63.71-13.62zM320 352c-44.19 0-80-42.99-80-96 0-53.02 35.82-96 80-96s80 42.98 80 96c0 53.03-35.83 96-80 96zm272 27.78c-17.52-4.39-35.71-6.85-54.32-8.44 5.87-26.08 27.5-45.88 54.32-49.28v57.72zm0-236.11c-30.89-3.91-54.86-29.7-55.81-61.55 19.54 2.17 38.09 6.23 55.81 12.66v48.89z"
          />
        </svg>
      </span>
      <h2>
        NASDAQ
      </h2>
    </div>
    <div
      className="search-bar"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
        <path
          d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
        />
      </svg>
      <input
        onChange={[Function]}
        placeholder="Search a company in NASDAQ"
        type="text"
      />
    </div>
    <h4>
      MARKET PRICE BY COMPANIES
    </h4>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
        }
      }
    />
  </div>
</div>
`;
